Code/Framework/AzCore/AzCore/Settings/SettingRegisteryUtility.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Settings/SettingsRegistry.h>
+
+namespace AZ::SettingsRegistryUtils
+{
+    template<typename T>
+    inline void SetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& value)
+    {
+        if (registry)
+        {
+            registry->Set(setting, AZStd::forward<T>(value));
+        }
+    }
+
+    template<typename T>
+    inline AZStd::remove_cvref_t<T> GetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& defaultValue)
+    {
+        AZStd::remove_cvref_t<T> value = AZStd::forward<T>(defaultValue);
+        if (registry)
+        {
+            T potentialValue;
+            if (registry->Get(potentialValue, setting))
+            {
+                value = AZStd::move(potentialValue);
+            }
+        }
+
+        return value;
+    }
+} // namespace AZ::SettingsRegistryUtils
```

Comment:
There are the same functions in [ViewportSettings.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportSettings.h#L15-L38), instead of adding this code, that file can just be included.

Also both of these functions have flaws, that the member Get/Set functions on the SettingsRegistry does not experience.

First this `SetRegistry` function doesn't return a boolean or `AZ::Outcome<void, AZStd::string>` indicating if the value was actually set
```
    using ErrorString = AZStd::fixed_string<128>;
    using RegistryOutcome = AZ::Outcome<void, ErrorString>;

    template<typename T>
    inline RegistryOutcome SetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& value)
    {
        if (registry == nullptr)
        {
             return AZ::Failure(ErrorString::format(R"(Failed to set registry entry for setting "%.*s". The Settings Registry is nullptr.)", AZ_STRING_ARG(setting));
        }
        if (!registry->Set(setting, AZStd::forward<T>(value))
        {
            return AZ::Failure(ErrorString::format(R"(Failed to set registry entry for setting "%.*s". The SettingsRegistry::Set has failed.)", AZ_STRING_ARG(setting));
        }

        return AZ::Success();
    }
```
Next the `GetRegistry1 function should defer assigning the defaultValue until after the query of the value from the SettingsRegistry fails to avoid an unneccessary assignment.

i.e
```
    template<typename T>
    inline AZStd::remove_cvref_t<T> GetRegistry(SettingsRegistryInterface* registry, const AZStd::string_view setting, T&& defaultValue)
    {
        if (registry == nullptr)
        {
            return AZStd::forward<T>(defaultValue);
        }

        if (AZStd::remove_cvref_t<T> value; !registry->Get(value, setting))
        {
            return AZStd::forward<T>(defaultValue);
        }

        return value;
    }
```

Also final note, the filename of "SettingsRegist**e**ryUtility.h" has an extra e in it.
The PR description also has an extra e in the term "SettingsRegist**e**ry"

Reply:
if were going to change the functionality of the utility method can just deprecate those methods under [ViewportSettings.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportSettings.h#L15-L38)

Reply:
The utility methods were just little helpers not intended to be used widely (that's why they weren't added to the Settings Registry core itself). They are all known values that we know will work which is why there's less error checking to make things less verbose and noisy. If we can keep a nice clean interface and update them to what @lumberyard-employee-dm suggested I'm fine with that but please don't make all the existing calls super noisy with lots of extra checks, that was the whole point of having these helper functions for the viewport settings in the first place.